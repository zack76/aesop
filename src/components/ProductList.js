import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import {Accordion, Card, Button} from "react-bootstrap";
import SubCategory from "./SubCategory"

const ProductList = ({ products }) => {
    return products.map((category, index) => (
        <Accordion key={index} defaultActiveKey="0">
            <Card>
                <Card.Header>
                    <Accordion.Toggle as={Button} variant="link" eventKey={'category' + index}>
                        {category.name}
                    </Accordion.Toggle>
                </Card.Header>
                <Accordion.Collapse eventKey={'category' + index}>
                    <Card.Body>
                        <SubCategory subcategories={category.items} />
                    </Card.Body>
                </Accordion.Collapse>
            </Card>
        </Accordion>
    ))
}

export default ProductList
