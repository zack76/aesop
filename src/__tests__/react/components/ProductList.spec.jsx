import React from 'react';
import { mount, shallow } from 'enzyme';
import ProductList from '../../../components/ProductList';
import { Accordion, Card, Button } from 'react-bootstrap';

jest.mock('../../../components/SubCategory', () => 'sub-category')
describe('ProductList', () => {
    const props = {
        products: [
            { name: 'toy', items: ['a', 'b', 'c'] },
            { name: 'paper', items: ['one', 'two', 'three'] },
            { name: 'food', items: ['apple', 'pear', 'orange'] }
        ]
    };

    it('Should render without error', () => {
        const wrapper = mount(<ProductList {...props} />);
        expect(wrapper.find(Accordion).length).toEqual(props.products.length);
        expect(
            wrapper
                .find(Accordion.Toggle)
                .first()
                .text()
        ).toBe('toy');
        expect(
            wrapper
                .find(Card.Body)
                .first()
                .length
        ).toEqual(1);
        expect(wrapper.find('sub-category').first().length).toEqual(1);
        expect(wrapper).toMatchSnapshot();
    })

    it('Should have Accordion.Toggle with "as" is Button', () => {
        const wrapper = shallow(<ProductList {...props} />);
        expect(
            wrapper
                .find(Accordion.Toggle)
                .first()
                .prop('as')
        ).toBe(Button);
        expect(wrapper).toMatchSnapshot();
    });
});
