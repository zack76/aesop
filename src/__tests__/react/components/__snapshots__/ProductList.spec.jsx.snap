// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Should have Accordion.Toggle with "as" is Button 1`] = `
Array [
  <ForwardRef
    defaultActiveKey="0"
    key="0"
  >
    <Card
      body={false}
    >
      <CardHeader>
        <ForwardRef
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            }
          }
          eventKey="category0"
          variant="link"
        >
          toy
        </ForwardRef>
      </CardHeader>
      <AccordionCollapse
        eventKey="category0"
      >
        <CardBody>
          <sub-category
            subcategories={
              Array [
                "a",
                "b",
                "c",
              ]
            }
          />
        </CardBody>
      </AccordionCollapse>
    </Card>
  </ForwardRef>,
  <ForwardRef
    defaultActiveKey="0"
    key="1"
  >
    <Card
      body={false}
    >
      <CardHeader>
        <ForwardRef
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            }
          }
          eventKey="category1"
          variant="link"
        >
          paper
        </ForwardRef>
      </CardHeader>
      <AccordionCollapse
        eventKey="category1"
      >
        <CardBody>
          <sub-category
            subcategories={
              Array [
                "one",
                "two",
                "three",
              ]
            }
          />
        </CardBody>
      </AccordionCollapse>
    </Card>
  </ForwardRef>,
  <ForwardRef
    defaultActiveKey="0"
    key="2"
  >
    <Card
      body={false}
    >
      <CardHeader>
        <ForwardRef
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            }
          }
          eventKey="category2"
          variant="link"
        >
          food
        </ForwardRef>
      </CardHeader>
      <AccordionCollapse
        eventKey="category2"
      >
        <CardBody>
          <sub-category
            subcategories={
              Array [
                "apple",
                "pear",
                "orange",
              ]
            }
          />
        </CardBody>
      </AccordionCollapse>
    </Card>
  </ForwardRef>,
]
`;

exports[`ProductList Should render without error 1`] = `
<ProductList
  products={
    Array [
      Object {
        "items": Array [
          "a",
          "b",
          "c",
        ],
        "name": "toy",
      },
      Object {
        "items": Array [
          "one",
          "two",
          "three",
        ],
        "name": "paper",
      },
      Object {
        "items": Array [
          "apple",
          "pear",
          "orange",
        ],
        "name": "food",
      },
    ]
  }
>
  <ForwardRef
    defaultActiveKey="0"
    key="0"
  >
    <div
      className="accordion"
    >
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <CardHeader>
            <div
              className="card-header"
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "type": "button",
                      "variant": "primary",
                    },
                    "displayName": "Button",
                    "render": [Function],
                  }
                }
                eventKey="category0"
                variant="link"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="link"
                >
                  <button
                    className="btn btn-link"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    toy
                  </button>
                </Button>
              </ForwardRef>
            </div>
          </CardHeader>
          <AccordionCollapse
            eventKey="category0"
          >
            <Collapse
              appear={false}
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                aria-expanded={null}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <CardBody>
                    <div
                      className="card-body"
                    >
                      <sub-category
                        subcategories={
                          Array [
                            "a",
                            "b",
                            "c",
                          ]
                        }
                      />
                    </div>
                  </CardBody>
                </div>
              </Transition>
            </Collapse>
          </AccordionCollapse>
        </div>
      </Card>
    </div>
  </ForwardRef>
  <ForwardRef
    defaultActiveKey="0"
    key="1"
  >
    <div
      className="accordion"
    >
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <CardHeader>
            <div
              className="card-header"
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "type": "button",
                      "variant": "primary",
                    },
                    "displayName": "Button",
                    "render": [Function],
                  }
                }
                eventKey="category1"
                variant="link"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="link"
                >
                  <button
                    className="btn btn-link"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    paper
                  </button>
                </Button>
              </ForwardRef>
            </div>
          </CardHeader>
          <AccordionCollapse
            eventKey="category1"
          >
            <Collapse
              appear={false}
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                aria-expanded={null}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <CardBody>
                    <div
                      className="card-body"
                    >
                      <sub-category
                        subcategories={
                          Array [
                            "one",
                            "two",
                            "three",
                          ]
                        }
                      />
                    </div>
                  </CardBody>
                </div>
              </Transition>
            </Collapse>
          </AccordionCollapse>
        </div>
      </Card>
    </div>
  </ForwardRef>
  <ForwardRef
    defaultActiveKey="0"
    key="2"
  >
    <div
      className="accordion"
    >
      <Card
        body={false}
      >
        <div
          className="card"
        >
          <CardHeader>
            <div
              className="card-header"
            >
              <ForwardRef
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "active": false,
                      "disabled": false,
                      "type": "button",
                      "variant": "primary",
                    },
                    "displayName": "Button",
                    "render": [Function],
                  }
                }
                eventKey="category2"
                variant="link"
              >
                <Button
                  active={false}
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                  variant="link"
                >
                  <button
                    className="btn btn-link"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    food
                  </button>
                </Button>
              </ForwardRef>
            </div>
          </CardHeader>
          <AccordionCollapse
            eventKey="category2"
          >
            <Collapse
              appear={false}
              dimension="height"
              getDimensionValue={[Function]}
              in={false}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                aria-expanded={null}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <CardBody>
                    <div
                      className="card-body"
                    >
                      <sub-category
                        subcategories={
                          Array [
                            "apple",
                            "pear",
                            "orange",
                          ]
                        }
                      />
                    </div>
                  </CardBody>
                </div>
              </Transition>
            </Collapse>
          </AccordionCollapse>
        </div>
      </Card>
    </div>
  </ForwardRef>
</ProductList>
`;
